diff --git a/.Makefile.bat b/.Makefile.bat
index aa3a722..429fc6b 100644
--- a/.Makefile.bat
+++ b/.Makefile.bat
@@ -9,14 +9,19 @@
 @SET target=-DTCC_TARGET_PE -DTCC_TARGET_I386
 @SET CC=mingw32-gcc 
 @SET P=32
-@GOTO tools
+@GOTO ut_tools
 
 :x86_64
 @SET target=-DTCC_TARGET_PE -DTCC_TARGET_X86_64
 @REM mingw 64 has an ICE with -Os
 @SET CC=x86_64-pc-mingw32-gcc -O0 -s -fno-strict-aliasing
 @SET P=64
-@GOTO tools
+@GOTO ut_tools
+
+:ut_tools
+@IF NOT EXIST include\nul mkdir include
+@copy ..\include\ut*.h include
+@IF %ERRORLEVEL% EQU 1 GOTO error
 
 :tools
 %CC% %target% tools/tiny_impdef.c -o tiny_impdef.exe 
diff --git a/tcc.h b/tcc.h
index 3f41917..9cbf1bf 100644
--- a/tcc.h
+++ b/tcc.h
@@ -356,8 +356,10 @@ typedef struct Section {
     struct Section *link;    /* link to another section */
     struct Section *reloc;   /* corresponding section for relocation, if any */
     struct Section *hash;     /* hash table for symbols */
+    struct Section *generic; /* generic data */
     struct Section *next;
     char name[1];           /* section name */
+
 } Section;
 
 typedef struct DLLReference {
diff --git a/tccgen.c b/tccgen.c
index 3a4a3f1..407751f 100644
--- a/tccgen.c
+++ b/tccgen.c
@@ -3906,8 +3906,12 @@ vpushi(type2.ref->v);
             }
             next();
         }else if (tok == ':' || tok == TOK_CL_ARROW) {
-            int qualifiers;
-				SValue *sv = vtop;
+           int qualifiers;
+           int bt, v, t;
+
+           SValue *sv = vtop;
+           t = sv->type.t & VT_TYPE;
+           bt = t & VT_BTYPE;
             /* field */ 
            if (tok == TOK_CL_ARROW) 
               indir();
diff --git a/win32/libtcc.dll b/win32/libtcc.dll
index 478a7e9..43c473e 100644
Binary files a/win32/libtcc.dll and b/win32/libtcc.dll differ
